app.js

```diff
@@ -1,16 +1,25 @@
 const clock = document.getElementById('clock')
 const startButton = document.getElementById('startButton')
+const stopButton = document.getElementById('stopButton')
 
 let seconds = 0
 
 let isRunning = false;
+let intervalID;
 
 startButton.addEventListener("click", () => {
     if (isRunning === false) {
-        setInterval(() => {
+        intervalID = setInterval(() => {
         seconds++;
         clock.textContent = seconds;
     }, 1000);
     isRunning = true;
     }
 });
+
+stopButton.addEventListener("click", () => {
+    if (isRunning === true) {
```

Comment:
is this check really necessary?

Reply:
I did put thought into whether it needed it or not. I put it there just as an act of diligence but I agree it doesn't really need it, just extra code right?

Reply:
it's an unnecessary check. Whether it's running or not we can carry out the same function calls with zero side effects.